File: Component/BusiLib/src/main/java-person/com/zq/person/adapter/ProducationAdapter.java
package com.zq.person.adapter;

import android.support.annotation.NonNull;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.common.view.recyclerview.DiffAdapter;
import com.component.busilib.R;
import com.zq.person.holder.EmptyProducationHolder;
import com.zq.person.holder.ProducationHolder;
import com.zq.person.model.ProducationModel;

/**
 * 用recycleview做一个空页面
 */
public class ProducationAdapter extends DiffAdapter<ProducationModel, RecyclerView.ViewHolder> {

    private static final int TYPE_NORMAL = 1;
    private static final int TYPE_EMPTY = 2;

    Listener mListener;
    boolean mIsSelf;

    int mProducationworksID = -1;  //选中播放的id

    LinearLayoutManager mLinearLayoutManager;

    public ProducationAdapter(Listener listener, boolean isSelf, LinearLayoutManager linearLayoutManager) {
        this.mIsSelf = isSelf;
        this.mListener = listener;
        this.mLinearLayoutManager = linearLayoutManager;
    }

    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        if (viewType == TYPE_NORMAL) {
            View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.producation_item_view_layout, parent, false);
            ProducationHolder viewHolder = new ProducationHolder(view, mListener, mIsSelf);
            return viewHolder;
        } else {
            View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.produacation_empty_view_layout, parent, false);
            EmptyProducationHolder viewHolder = new EmptyProducationHolder(view);
            return viewHolder;
        }
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int position) {
        if (mDataList != null && mDataList.size() > 0) {
            ProducationModel model = mDataList.get(position);
            ProducationHolder viewHolder = (ProducationHolder) holder;
            if (mProducationworksID == model.getWorksID()) {
                viewHolder.bindData(position, model, true);
            } else {
                viewHolder.bindData(position, model, false);
            }
        } else {
            EmptyProducationHolder viewHolder = (EmptyProducationHolder) holder;
            viewHolder.bindData(mIsSelf);
        }
    }

    @Override
    public int getItemCount() {
        if (mDataList != null && mDataList.size() > 0) {
            return mDataList.size();
        } else {
            return 1;
        }
    }

    @Override
    public int getItemViewType(int position) {
        if (mDataList != null && mDataList.size() > 0) {
            return TYPE_NORMAL;
        } else {
            return TYPE_EMPTY;
        }
    }

    public int getPlayingWorksIdPosition() {
        return mProducationworksID;
    }

    public void setPlayPosition(int workId) {
        mProducationworksID = workId;
        notifyDataSetChanged();
    }

    public interface Listener {

        void onClickDele(int position, ProducationModel model);

        void onClickShare(int position, ProducationModel model);

        void onClickPlayBtn(View view, boolean play, int position, ProducationModel model);
    }
}

File: Component/BusiLib/src/main/java-person/com/component/person/producation/adapter/ProducationAdapter.kt
package com.component.person.producation.adapter

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import com.common.view.recyclerview.DiffAdapter
import com.component.busilib.R
import com.component.person.producation.model.ProducationModel
import com.component.person.producation.holder.EmptyProducationHolder
import com.component.person.producation.holder.ProducationHolder

/**
 * 用recycleview做一个空页面
 */
class ProducationAdapter(private var mIsSelf: Boolean) : DiffAdapter<ProducationModel, RecyclerView.ViewHolder>() {

    var playingWorksIdPosition = -1
        internal set  //选中播放的id

    var mOnClickDeleListener: ((position: Int, model: ProducationModel?) -> Unit)? = null
    var mOnClickShareListener: ((position: Int, model: ProducationModel?) -> Unit)? = null
    var mOnClickPlayListener: ((view: View, play: Boolean, position: Int, model: ProducationModel?) -> Unit)? = null

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        if (viewType == TYPE_NORMAL) {
            val view = LayoutInflater.from(parent.context).inflate(R.layout.producation_item_view_layout, parent, false)
            return ProducationHolder(view, mIsSelf, mOnClickDeleListener, mOnClickShareListener, mOnClickPlayListener)
        } else {
            val view = LayoutInflater.from(parent.context).inflate(R.layout.produacation_empty_view_layout, parent, false)
            return EmptyProducationHolder(view)
        }
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        if (mDataList != null && mDataList.size > 0) {
            val model = mDataList[position]
            val viewHolder = holder as ProducationHolder
            if (playingWorksIdPosition == model.worksID) {
                viewHolder.bindData(position, model, true)
            } else {
                viewHolder.bindData(position, model, false)
            }
        } else {
            val viewHolder = holder as EmptyProducationHolder
            viewHolder.bindData(mIsSelf)
        }
    }

    override fun getItemCount(): Int {
        return if (mDataList != null && mDataList.size > 0) {
            mDataList.size
        } else {
            1
        }
    }

    override fun getItemViewType(position: Int): Int {
        return if (mDataList != null && mDataList.size > 0) {
            TYPE_NORMAL
        } else {
            TYPE_EMPTY
        }
    }

    fun setPlayPosition(workId: Int) {
        playingWorksIdPosition = workId
        notifyDataSetChanged()
    }

    companion object {

        private val TYPE_NORMAL = 1
        private val TYPE_EMPTY = 2
    }
}
